compliance_tool/cli.py

```diff
@@ -1,57 +1,79 @@
 import argparse
-from compliance_tool import list, scan, remediate
+from compliance_tool import list_, scan, remediate
 
 
-def prepare_parser_list(subparsers) -> None:
-    parser = subparsers.add_parser("list",
-                                   help="list available components")
-    parser.set_defaults(func=list.execute)
-    list_subparsers = parser.add_subparsers(required=True)
+def prepare_parser_list(cli_sub_parsers, global_parser, profile_parser) -> None:
+    parser = cli_sub_parsers.add_parser("list", parents=[global_parser],
+                                        help="list available components"
+                                             " (profiles, rules, controls),"
+                                             " will list profiles if component type"
+                                             " is not specified")
+    # TODO: Set individual functions for sub-parsers to distinguish between listing modes
+    #       this one should be linked with listing profiles as we consider it to
+    #       be the default listing operation.
+    parser.set_defaults(func=list_.execute)
+    list_subparsers = parser.add_subparsers(required=False)
 
     list_subparsers.add_parser("profiles",
                                help="list available profiles")
 
-    parser = list_subparsers.add_parser("controls",
-                                        help="list available controls for a given profile")
-    parser.add_argument("--profile")
+    list_subparsers.add_parser("controls", parents=[profile_parser],
```

Comment:
In the specification doc we use the term `requirements` but here is used `controls`. We should stick to the specification doc.